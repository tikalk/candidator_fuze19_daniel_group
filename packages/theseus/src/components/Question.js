import React from 'react';
import PropTypes from 'prop-types';
import { markdown } from 'markdown';
import styled from 'styled-components';
import Timer from './Timer';
import Colors from '../styles/Colors';

const QuestionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  padding-bottom: 20px;
`;

const HeroImage = styled.img`
  max-width: 20%;
  max-height: 200px;
`;

const QuestionText = styled.div`
  width: 100%;
  text-align: left;
  overflow-y: auto;
  padding-right: 2rem;
`;

const Attempts = styled.div`
  color: ${props => (props.attempts === props.maxAttempts ? Colors.warnRed : 'inherit')};
`;

function Question({ question, onTimeUpdate, attempts, onTimeUp }) {
  return (
    <div>
      <QuestionContainer>
        <QuestionText
          dangerouslySetInnerHTML={{
            __html: markdown.toHTML(question.description),
          }}
        />
        <HeroImage src={question.heroImage} />
      </QuestionContainer>
      <Attempts attempts={attempts} maxAttempts={question.attempts}>
        Attempts: {attempts}/{question.attempts}
      </Attempts>
      <Timer
        key={question._id}
        onTimeUpdate={onTimeUpdate}
        onTimeUp={onTimeUp}
        label="Time Left"
        minutes={question.time.minutes}
        seconds={question.time.seconds}
      />
    </div>
  );
}

Question.propTypes = {
  onTimeUpdate: PropTypes.func,
  onTimeUp: PropTypes.func,
  question: PropTypes.shape({
    description: PropTypes.string.isRequired,
    heroImage: PropTypes.string.isRequired,
  }).isRequired,
};

export default Question;
