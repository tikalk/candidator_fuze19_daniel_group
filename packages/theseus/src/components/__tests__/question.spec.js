import React from 'react';
import renderer from 'react-test-renderer';
import Question from '../Question';

describe('Test <Question>', () => {
  it('should render correctly', () => {
    const question = {
      description: 'description...',
      heroImage: 'http://url-of-hero.image',
    };
    const tree = renderer.create(<Question question={question} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
