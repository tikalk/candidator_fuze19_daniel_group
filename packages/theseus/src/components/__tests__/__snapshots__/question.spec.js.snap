// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Question> should render correctly 1`] = `
<div
  className="sc-bdVaJa iqBxLi"
>
  <div
    className="sc-htpNat hvVCPt"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>description...</p>",
      }
    }
  />
  <img
    className="sc-bwzfXH ijDTxx"
    src="http://url-of-hero.image"
  />
</div>
`;
